### PR TITLE
bootstrap-coa-env(*)

### LINKED ISSUE #113: provide automated non-attended COA env bootstrap
User Story outline:
* As a COA contributor, in order to run end-to-end automated tests, I need to set up minimum COA prereq environment
* As a paas-template author, in order to contribute templates, I need to set up COA prereq minimum environment
* As a paas-template operator, in order to deploy paas-template, I need to bootstrap COA prereqs

Acceptance scenario:
- Given raw Iaas prereqs (openstack credentials)
- When I launch the COA bootstrap
- Then transient infrastructure (bosh director / concourse / credhub) is set up
- And [paas-templates](https://github.com/orange-cloudfoundry/paas-templates) micro-deps deployment subset gets deployed with
   - micro-depls/micro-bosh
   - micro-credhub
   - micro-dns
   - micro-minio
   - concourse-micro
   - gitlab + config

### PR DESCRIPTION
**Pull request by lucaspinto**
Fixes #113  .

Changes proposed in this pull-request:
- adds a script called "bootstrap_coa_env.rb" that, when fed a list of prerequisites credentials, creates a minimal COA environment.

### CODE REVIEW

File: docs/reference_dataset/pipelines/hello-world-root-depls-generated.yml [hunk 1]
```diff
@@ -75,7 +75,7 @@ resources:
     bucket: ((s3-stemcell-bucket))
     region_name: ((s3-stemcell-region-name))
     # customization is required to remove bosh prefix in stemcell name
-    regexp: ((stemcell-name-prefix))((stemcell-main-name))/bosh-stemcell-(.*)-((stemcell-main-name)).tgz
```
Comment (lucaspinto) [inline docs/reference_dataset/pipelines/hello-world-root-depls-generated.yml]: We may want to remove this.

File: spec/helpers/utils.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+require 'securerandom'
+
+def fixtures_dir(path = '')
+  File.join(File.join(File.dirname(__FILE__), '..', path, 'fixtures'))
+end
+
+def new_tmpdir_path
```
Comment (o-orand) [inline spec/helpers/utils.rb]: Why not using mktmpdir ?, so we won't need tmp/.gitkeep

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -133,5 +135,4 @@ def fly(arg, env = {})
   def execute(cmd, env = {})
     fly("execute #{cmd}", env)
   end
-
 end
```
Comment (o-orand) [inline spec/spec_helper.rb]: Is this really useful `tmp/.gitkeep`

**Comment by o-orand** [comment]
to fix the stemcells issue, we need the online stemcell support (#128)

**Comment by lucaspinto** [comment]
I updated the branch.

**Comment by o-orand** [comment]
Could you rebase and squash both commits into one.

File: scripts/extract_pipeline_credentials_list.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+def credential_list(pipelines_path)
+  creds_list = []
+
+  files = File.directory?(pipelines_path) ? Dir["#{pipelines_path}/*"] : [pipelines_path]
+
+  files.each do |path|
+    next if File.directory?(path)
+    pipeline_content = File.read(path)
+    creds_list << pipeline_content.scan(/\(\(([\w|-]*)\)\)/).flatten.uniq
```
Comment (o-orand) [inline scripts/extract_pipeline_credentials_list.rb]: please use %r{ } syntax

**Comment by lucaspinto** [comment]
I still need to fix the docs.

**Comment by lucaspinto** [comment]
Docs fixed

**Comment by o-orand** [comment]
We still need to work on running bootstrap on openstack

**Pull request merged**